src/graph/DependencyGraph.cpp

```diff
@@ -82,6 +82,56 @@ size_t DependencyGraph::addNode()
   return roots_.size() - 1;
 }
 
+void DependencyGraph::removeNode(size_t id)
+{
+  using Edge = std::pair<size_t, size_t>;
+  assert(id < roots_.size());
+  // Remove all edges from/to {id}
+  for(auto it = edges_.begin(); it != edges_.end();)
+  {
+    const Edge & e = *it;
+    if(e.first == id || e.second == id)
+    {
+      it = edges_.erase(it);
+    }
+    else
+    {
+      ++it;
+    }
+  }
+  // We check if the children of {id} can be promoted to roots_
+  // i.e. if they have no more edges pointing to them
+  for(auto c : children_[id])
+  {
+    if(std::find_if(edges_.begin(), edges_.end(), [&](const Edge & e) { return e.second == c; }) == edges_.end())
```

Comment:
Note that if edges are properly updated, the comparison needs to be a little different `return c > id ? (e.second + 1 == c) : (e.second == c);`

Reply:
Fixed in cebeda8379ebf7d9d8be1b6a8d00322f692bf440